src/main/host/descriptor/tcp.c

```diff
@@ -1580,7 +1580,7 @@ void tcp_getInfo(TCP* tcp, struct tcp_info *tcpinfo) {
     tcpinfo->tcpi_snd_ssthresh = (u_int32_t)tcp->cong.hooks->tcp_cong_ssthresh(tcp);
     tcpinfo->tcpi_snd_cwnd = (u_int32_t)tcp->cong.cwnd;
     tcpinfo->tcpi_advmss =
-        (u_int32_t)(CONFIG_MTU - CONFIG_HEADER_SIZE_TCPIP - CONFIG_HEADER_SIZE_ETH);
+        (u_int32_t)(CONFIG_MTU - CONFIG_HEADER_SIZE_TCPIP);
```

Comment:
Consider using the CONFIG_TCP_MAX_SEGMENT_SIZE constant defined in src/main/core/support/definitions.h.